services/DropSubmissionService.ts

```diff
@@ -148,6 +148,118 @@ export const reactWithBasePoints = async (message: Message) => {
     await message.react(NumberEmojis.TEN);
 };
 
+export const processNonembedDinkPost = async (message: Message, pointsSheetLookup: Record<string, string>) => {
+    // Handle submissions that are not embeded
+    console.debug(message.content);
+
+    if (message.channel.type !== ChannelType.GuildText) {
+        console.error("The message was not posted in the correct channel: ", message.channel.type);
+        return;
+    }
+
+    const matches = message.content.match(/\*\*([\s\S]*?)\*\*/g);
+    console.debug("Matches: ", matches);
+
+    if (matches === null) {
+        console.error("No bold text found in the message content: ", message.content);
+        return;
+    }
+
+    const pieces = matches.map(part => part.replace(/\*/g, '')).filter(part => (part.length > 0 && !part.includes('*')));
+    console.debug("Pieces: ", pieces);
+
+    if (pieces.length === 1 && message.content.includes("** just got a pet!")) {
+        // Add "1 x " & " ()" so it matches the expected input later
+        pieces[1] = "1 x Pet ()";
+        // Could add a lookup here to get the source of the pet
+        // The "Pet" section of dink does not have the %SOURCE% parameter available
+        pieces[2] = "unknown";
+    } else if (pieces.length !== 3) {
+        console.error("The message did not return 3 pieces: ", pieces);
+        return;
+    }
+
+    const player = pieces[0];
+    const loot = pieces[1].split("\n").map(piece => ({
+        'item': piece.substring(piece.indexOf(" ", 2) + 1, piece.lastIndexOf(" ")),
+        'quantity': parseInt(piece.substring(0, piece.indexOf(" ", 1)))
+    }));
+    const source = pieces[2];
+
+    // console.debug("Player: ", player);
+    // console.debug("Loot: ", loot);
+    // console.debug("Source: ", source);
+    
+    const allUsers = await message?.guild?.members.fetch();
+    const possibleUser = allUsers?.find((x) => (x.nickname ?? '').toLocaleLowerCase().startsWith(player.toLocaleLowerCase()));
+
+    // console.debug("Possible user: ", possibleUser);
+    
+    if (!possibleUser) {
+        console.error("No possible user found: ", possibleUser);
+        return;
+    }
+
+    if (loot.length === 0) {
+        console.error("No loot found: ", loot);
+        return;
+    }
+
+    const valid_loot: Array<{name: string; points: number; quantity: number}> = [];
+    let points_to_add = 0;
+
+    for (let i = 0; i < loot.length; ++i) {
+        const item_name = loot[i].item.toLocaleLowerCase();
+        if (!(item_name in pointsSheetLookup)) {
+            console.warn(`${loot[i].item} (${item_name}) does not exist on the list of items that are eligible!`);
+            continue;
+        }
+
+        const points = parseInt(pointsSheetLookup[item_name]);
+
+        points_to_add += points * loot[i].quantity;
+        valid_loot.push({
+            name: loot[i].item,
+            points: points,
+            quantity: loot[i].quantity
+        });
+    }
```

Comment:
```suggestion
       const validLoot = loot
        .filter(x => x.item.toLocaleLowerCase() in pointsSheetLookup)
        .map((itemWorthPoints): {name: string; points: number; quantity: number} => ({
            name: itemWorthPoints.item,
            points: parseInt(pointsSheetLookup[itemWorthPoints.item.toLocaleLowerCase()], 10),
            quantity: itemWorthPoints.quantity
        }));

    const totalPointsToAdd = validLoot.reduce((total, item) => total + (item.points * item.quantity), 0);
```

Reply:
if you wouldn't mind, could you try using this instead? I try to use function-style programming where possible and avoid mutation at all cost:
```ts
    const validLoot = loot
        .filter(x => x.item.toLocaleLowerCase() in pointsSheetLookup)
        .map((itemWorthPoints): {name: string; points: number; quantity: number} => ({
            name: itemWorthPoints.item,
            points: parseInt(pointsSheetLookup[itemWorthPoints.item.toLocaleLowerCase()], 10),
            quantity: itemWorthPoints.quantity
        }));

    const totalPointsToAdd = validLoot.reduce((total, item) => total + (item.points * item.quantity), 0);
    ```

Reply:
and then update the replace the variables (also prefer camelCase instead of snake_case)